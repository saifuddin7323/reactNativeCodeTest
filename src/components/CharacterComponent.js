import React, {memo} from 'react';
import {View, StyleSheet, Text, Image} from 'react-native';

const styles = StyleSheet.create({
  containerStyle: {
    padding: 8,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  titleStyleBlue: {
    color: 'skyblue',
    fontWeight: 'bold',
  },
  titleStyleBlack: {
    color: 'black',
    fontWeight: 'bold',
    flex: 1,
    flexWrap: 'wrap',
  },
  imageStyle: {height: 80, width: 80, borderRadius: 40},
  flexRowStyle: {flexDirection: 'row'},
  flexOne: {flex: 1},
});

export const CharacterComponent = memo(
  ({
    fullName,
    title,
    imageUrl,
    containerStyle,
    imageStyle,
    headingColor,
    textColor,
  }) => {
    return (
      <View style={[styles.containerStyle, containerStyle]}>
        <View style={styles.flexOne}>
          <View style={styles.flexRowStyle}>
            <Text
              style={[
                styles.titleStyleBlue,
                headingColor && {color: headingColor},
              ]}>
              {'Name'}:{' '}
            </Text>
            <Text
              style={[styles.titleStyleBlack, textColor && {color: textColor}]}>
              {fullName?.toUpperCase()}
            </Text>
          </View>
          <View style={styles.flexRowStyle}>
            <Text
              style={[
                styles.titleStyleBlue,
                headingColor && {color: headingColor},
              ]}>
              {'Title'}:{' '}
            </Text>
            <Text
              style={[styles.titleStyleBlack, textColor && {color: textColor}]}>
              {title?.toUpperCase()}
            </Text>
          </View>
        </View>
        <Image
          style={[styles.imageStyle, imageStyle]}
          source={{uri: imageUrl}}
        />
      </View>
    );
  },
);
