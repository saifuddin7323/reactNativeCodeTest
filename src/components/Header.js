import React, {memo} from 'react';
import {View, StyleSheet, Text} from 'react-native';

const styles = StyleSheet.create({
  containerStyle: {
    backgroundColor: 'skyblue',
    padding: 8,
    justifyContent: 'center',
    height: 50,
  },
  titleStyle: {color: 'white', fontSize: 24, textAlign: 'center'},
});

export const Header = memo(() => {
  return (
    <View style={styles.containerStyle}>
      <Text style={styles.titleStyle}>{'Home'}</Text>
    </View>
  );
});
