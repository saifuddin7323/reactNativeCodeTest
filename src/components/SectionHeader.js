import React, {memo} from 'react';
import {View, StyleSheet, Text} from 'react-native';

const styles = StyleSheet.create({
  containerStyle: {flexDirection: 'row', backgroundColor: 'black', padding: 8},
  titleStyle: {color: 'white', fontSize: 18, fontWeight: 'bold'},
});

export const SectionHeader = memo(({title, length}) => {
  return (
    <View style={styles.containerStyle}>
      <Text style={styles.titleStyle}>{title}</Text>
      <Text style={styles.titleStyle}>{': ' + length}</Text>
    </View>
  );
});
