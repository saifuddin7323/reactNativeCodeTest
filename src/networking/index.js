import {getCharacterData} from './GetCharacter';
export {getCharacterData};
