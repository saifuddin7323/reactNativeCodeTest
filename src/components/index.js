import {CharacterComponent} from './CharacterComponent';
import {SectionHeader} from './SectionHeader';
import {Header} from './Header';
export {CharacterComponent, SectionHeader, Header};
