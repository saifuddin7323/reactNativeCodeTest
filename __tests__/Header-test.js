/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {Header} from '../src/components';

it('renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});
