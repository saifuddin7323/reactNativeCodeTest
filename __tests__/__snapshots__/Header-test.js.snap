// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "skyblue",
      "height": 50,
      "justifyContent": "center",
      "padding": 8,
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "textAlign": "center",
      }
    }
  >
    Home
  </Text>
</View>
`;
